Packs/EWS/Integrations/EWSv2/EWSv2.py

```diff
@@ -1241,22 +1250,40 @@ def get_entry_for_file_attachment(item_id, attachment):
 
 def parse_attachment_as_dict(item_id, attachment):
     try:
-        attachment_content = attachment.content if isinstance(attachment,
-                                                              FileAttachment) else attachment.item.mime_content
-        return {
-            ATTACHMENT_ORIGINAL_ITEM_ID: item_id,
-            ATTACHMENT_ID: attachment.attachment_id.id,
-            'attachmentName': get_attachment_name(attachment.name),
-            'attachmentSHA256': hashlib.sha256(attachment_content).hexdigest() if attachment_content else None,
-            'attachmentContentType': attachment.content_type,
-            'attachmentContentId': attachment.content_id,
-            'attachmentContentLocation': attachment.content_location,
-            'attachmentSize': attachment.size,
-            'attachmentLastModifiedTime': attachment.last_modified_time.ewsformat(),
-            'attachmentIsInline': attachment.is_inline,
-            ATTACHMENT_TYPE: FILE_ATTACHMENT_TYPE if isinstance(attachment, FileAttachment) else ITEM_ATTACHMENT_TYPE
-        }
-    except TypeError, e:
+        # if this is a file attachment or a non-empty email attachment
+        if isinstance(attachment, FileAttachment) or hasattr(attachment, 'item'):
```

Comment:
what about `ItemAttachment`

Reply:
If it is an ItemAttachment we want to make sure it has 'item' attribute - hence it is covered by the second condition.